samples/simd/simd_lib.h

```diff
@@ -10,25 +10,14 @@
 extern C {
 #endif
 
-//typedef __attribute__ ((__vector_size__ (8))) float simdFloat2;
-//simdFloat2 getSimd8(float f1, float f2);
-//float length(simdFloat2);
-
 //uint8_t __attribute__ ((__vector_size__ (16))) cv_var;
 //unsigned char __attribute__ ((__vector_size__ (16))) cv2;
 //signed char __attribute__ ((__vector_size__ (16))) cv3;
 //char __attribute__ ((__vector_size__ (16))) cv31;
 //char volatile const __attribute__ ((__vector_size__ (16))) cv4;
 
-//typedef __attribute__((__ext_vector_type__(2))) float simd_float2;
 typedef __attribute__((__ext_vector_type__(2))) float simd_float2;
-//typedef __attribute__((__ext_vector_type__(2))) const float simd_float2_c;
-simd_float2 sf2;
-
-typedef const float CFloat;
-typedef CFloat __attribute__ ((__vector_size__ (16))) CFloatV;
-CFloatV const CFloat_var = {1., 3.162, 1., 31.};
-const float __attribute__ ((__vector_size__ (16)))    CFloat_var2  = {1., 3.162, 1., 31.};
+simd_float2 sf2;  // not supported
```

Comment:
You mean top level globals are not supported?

Reply:
Nope. The canonical type, which is gcc extension  __ext_vector_size__, is not supported. Clang provides it with kind = CXType_Unexposed instead of CXType_Vector. This should be investigated further, because clang actually supports this extension internally.